builder/dockerfile/dispatchers.go

```diff
@@ -71,6 +71,10 @@ func env(b *Builder, args []string, attributes map[string]bool, original string)
 			return errBlankCommandNames("ENV")
 		}
 
+		if runtime.GOOS == "windows" {
+			// Case insensitive environment variables on Windows
+			args[j] = strings.ToUpper(args[j])
```

Comment:
It does change the actual env key also, right? Not only the comment. If this is true then it can break programs that don't just use `getenv` but look at the keys directly. The fix should actually be in https://github.com/docker/docker/pull/28725/files#diff-77aa50fa42bcdf6c3bedc962a396b6eaR84 then.